### PR TITLE
common: add Base ETH send pair, chain/token cleanup

### PR DESCRIPTION
**Pull request by dcposch**
## Screenshots

image
image
image
image
image

## Known issues

The swap memo ($x -> Y ETH) doesn't appear while the transaction is pending. We can fix this in a future cleanup refactor.

## Release plan

- [x] Test locally
- [x] Deploy prod API
- [x] Create TestFlight build 
- [x] Once deployed successfully, test following link. **Send Vitalik $0.20 worth of ETH:** https://daimo.com/l/request?to=vitalik.eth&n=0.20&c=8453&t=eth
- [x] Ship

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/send/SendTransferButton.tsx [hunk 1]
```diff
@@ -95,10 +98,9 @@ export function SendTransferButton({
         chainGasConstants: account.chainGasConstants,
       };
 
-      // TODO: handle case with swap and bridge
-      assert(toCoin.symbol === "USDC");
-
       if (toChain.chainId !== account.homeChainId) {
+        // TODO: handle case with swap and bridge
```
Comment (kayleegeorge) [inline apps/daimo-mobile/src/view/screen/send/SendTransferButton.tsx:102]: 👍

File: apps/daimo-mobile/src/view/screen/send/SendTransferScreen.tsx [hunk 2]
```diff
@@ -101,10 +100,11 @@ function SendScreenInner({
     else goHome();
   }, [nav, money, recipient]);
 
-  const defaultHomeCoin = baseUSDC; // TODO: add homecoin in Account
-  const defaultHomeChain = getDAv2Chain(account.homeChainId);
-  toCoin = toCoin ?? defaultHomeCoin;
-  toChain = toChain ?? getDAv2Chain(account.homeChainId);
+  const { homeChainId, homeCoinAddress } = account;
```
Comment (kayleegeorge) [inline apps/daimo-mobile/src/view/screen/send/SendTransferScreen.tsx:103]: 🔥

File: packages/daimo-common/src/sendPair.ts [hunk 3]
```diff
@@ -27,11 +29,13 @@ const supportedSendPairsMainnet: SendPair[] = [
   { chain: optimism, coin: optimismUSDC },
   { chain: polygon, coin: polygonUSDC },
   { chain: arbitrum, coin: arbitrumUSDC },
+  { chain: base, coin: baseETH },
 ];
 
 const supportedSendPairsTestnet: SendPair[] = [
   { chain: baseSepolia, coin: baseSepoliaUSDC },
   { chain: ethereumSepolia, coin: ethereumSepoliaUSDC },
+  { chain: baseSepolia, coin: baseSepoliaETH },
```
Comment (kayleegeorge) [inline packages/daimo-common/src/sendPair.ts:38]: 🔥

File: packages/daimo-contract/src/foreignToken.ts [hunk 4]
```diff
@@ -362,56 +379,80 @@ export const avalancheFujiWAVAX: ForeignToken = {
...
+  const ret = tokensByChainId.get(chainId);
+  if (ret == null) throw new Error(`Unsupported chain ${chainId}`);
+  return ret;
 }
 
-export function getForeignCoinBySymbolAndChain(
-  symbol: string,
-  chainId: number
+/** Get a given token */
+export function getTokenByAddress(
```
Comment (kayleegeorge) [inline packages/daimo-contract/src/foreignToken.ts:451]: 🔥

File: apps/daimo-mobile/src/storage/storedTypes.ts [hunk 5]
```diff
@@ -233,7 +233,6 @@ export interface StoredV16ProposedSwap {
 interface StoredV16ForeignCoin {
   chainId: number;
   token: Address;
-  name: string;
```
Comment (kayleegeorge) [inline apps/daimo-mobile/src/storage/storedTypes.ts:236]: just checking this won't break backcompat right

Comment (dcposch) [reply apps/daimo-mobile/src/storage/storedTypes.ts:236]: yeah great question. i thought about it & i think it's fine in this case.

in general you're absolutely right, we should not change existing Stored... models

**Comment by dcposch** [comment]
image

**Pull request merged**